src/gbt.cpp

```diff
@@ -85,124 +65,88 @@ bool scanGBTPhasesLocal(localArgs *la, const uint32_t ohN, const uint32_t N, con
         }
     }
 
-    // Write the results to RPC keys
-    for (uint32_t vfatN = 0; vfatN < oh::VFATS_PER_OH; vfatN++) {
-        la->response->set_word_array(stdsprintf("OH%u.VFAT%u", ohN, vfatN), results[vfatN]);
-    }
+    return results;
 
-    return false;
-} //End scanGBTPhaseLocal
+} //End scanGBTPhase
 
-void writeGBTConfig(const RPCMsg *request, RPCMsg *response)
+bool writeGBTConfig::operator()(const uint32_t &ohN, const uint32_t &gbtN) const
 {
-    GETLOCALARGS(response);
-
-    // Get the keys
-    const uint32_t ohN = request->get_word("ohN");
-    const uint32_t gbtN = request->get_word("gbtN");
-
-    // We must check the size of the config key
-    const uint32_t configSize = request->get_binarydata_size("config");
-    if (configSize != gbt::CONFIG_SIZE)
-        EMIT_RPC_ERROR(response, stdsprintf("The provided configuration has not the correct size. It is %u registers long while this methods expects %hu 8-bits registers.", configSize, gbt::CONFIG_SIZE), (void)"");
 
-    gbt::config_t config{};
-    request->get_binarydata("config", config.data(), config.size());
```

Comment:
Here is where the configuration data was read from the RPC message and converted into the `gbt::config_t` type object

Reply:
Then since we're taking away the RPC message, how will I get the configuration data? will it just be explicitly input?

Reply:
Oh wait, nevermind, that is what you said earlier. I will fix it.